packages/prettier-plugin-java/src/printers/printer-utils.js

```diff
@@ -443,6 +443,52 @@ function isShiftOperator(tokens, index) {
   return "none";
 }
 
+function retrieveNodesToken(ctx) {
```

Comment:
This recursive re-collecting of all the tokens happens for **every** ignore-comment?
This does not seem to be very effective, but maybe it is o.k. because there should be very few ignore comments in most input sources...

I also think you only need the first and last tokens from a CSTNode (specifically their start/end offsets) so perhaps you do not really have to collect all the tokens just the min/max offsets?


But do not we already have the CSTNodes start/end offsets in the new location property? so perhaps this logic is redundant?

Reply:
> This recursive re-collecting of all the tokens happens for every ignore-comment?
> This does not seem to be very effective, but maybe it is o.k. because there should be very few ignore comments in most input sources...

I am using this function to retrieve all the tokens as we might need it for other functionnality. I am more thinking about organising imports. We will need this kind of function as we want to be sorted by the name of it.

> I also think you only need the first and last tokens from a CSTNode (specifically their start/end offsets) so perhaps you do not really have to collect all the tokens just the min/max offsets?

I agree, I was just using the function already written I can make another one to just retrieve the first and last token.

> But do not we already have the CSTNodes start/end offsets in the new location property? so perhaps this logic is redundant?

We do have the location but as I said on another conversation, the location on nodes does not include comments.
For instance: 
```java
// prettier-ignore
int a; // my int
```
The location on the CSTNode associated to ```int a;```` will start on ```i``` and end on the semicolon ```;``` and won't cover ```// my int``` so if I cut it won't cut the comment.

Reply:
O.k. I understand why you are doing this function now, just be wary about re-creating an augmented token vector with contents you do not actually need.

It is is highly inefficient...

Reply:
I am now using a function that avoid to rebuild a token vector.